aws_lambda_builders/workflows/go_dep/DESIGN.md

```diff
@@ -0,0 +1,33 @@
+# Go Dep - Lambda Builder
+
+## Scope
+Building Go projects using the dep tool (https://github.com/golang/dep) is rather simple, if you was to do
+this by hand, you would perform these commands:
+
+    - `dep ensure`
+    - `GOOS=linux GOARCH=amd64 go build -o handler main.go`
+    - `zip -r source.zip`
+
+The scope of `sam build` would be to create a macro for these commands to ensure that spelling and paths are correct.
```

Comment:
Can you make this the scope of the aws-lambda-builders? I want to make sure we focus on the this library provides directly so we don't constrain the library to only support SAM CLI.